paper-server/src/main/java/io/papermc/paper/command/VersionCommand.java

```diff
@@ -0,0 +1,160 @@
+package io.papermc.paper.command;
+
+import com.destroystokyo.paper.PaperVersionFetcher;
+import com.destroystokyo.paper.util.VersionFetcher;
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.arguments.StringArgumentType;
+import com.mojang.brigadier.builder.LiteralArgumentBuilder;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.suggestion.Suggestions;
+import com.mojang.brigadier.suggestion.SuggestionsBuilder;
+import io.papermc.paper.plugin.configuration.PluginMeta;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.JoinConfiguration;
+import net.kyori.adventure.text.event.ClickEvent;
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.format.TextDecoration;
+import net.kyori.adventure.text.serializer.plain.PlainTextComponentSerializer;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.Commands;
+import org.bukkit.Bukkit;
+import org.bukkit.command.CommandSender;
+import org.bukkit.plugin.Plugin;
+import org.bukkit.util.StringUtil;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
+
+@NullMarked
+public class VersionCommand {
+    private static final VersionFetcher versionFetcher = new PaperVersionFetcher();
+    private static final Set<CommandSender> versionWaiters = new HashSet<>();
+
+    private static @Nullable Component versionMessage = null;
+    private static boolean versionTaskStarted = false;
+    private static long lastCheck = 0;
+
+    public static void register(CommandDispatcher<CommandSourceStack> dispatcher) {
```

Comment:
Why this method for register commands? the rest of classes just depends on PaperCommands unless in future the team maybe wanna move to brigadier...